src/base/transport.ts

```diff
@@ -0,0 +1,130 @@
+import OAuth from 'oauth';
+import IClientOptions from './IClientOptions';
+import Cache from './Cache';
+import { formatURL } from './utils';
+
+class Transport {
+  private oauth: OAuth.OAuth;
+
+  private cache?: Cache;
+  private credentials: IClientOptions & { [key: string]: any };
+  constructor(options: IClientOptions) {
+    this.credentials = options;
+    this.oauth = new OAuth.OAuth(
+      'https://api.twitter.com/oauth/request_token',
+      'https://api.twitter.com/oauth/access_token',
+      this.credentials.apiKey,
+      this.credentials.apiSecret,
+      '1.0A',
+      null,
+      'HMAC-SHA1'
+    );
+    if (!options?.disableCache) {
+      this.cache = new Cache(options?.ttl, options.maxByteSize);
+    }
+  }
+  updateOptions(options: Partial<IClientOptions>) {
+    const { apiKey, apiSecret, ...rest } = options;
+    const cleanOptions = rest as { [key: string]: any };
+
+    Object.keys(cleanOptions).forEach((key: string) => {
+      if (cleanOptions[key]) {
+        this.credentials[key] = cleanOptions[key];
+      }
+    });
+  }
+  async doGetRequest<T>(url: string): Promise<T> {
```

Comment:
Empty line + public :slightly_smiling_face: 

```suggestion
  public async doGetRequest<T>(url: string): Promise<T> {
```

Reply:
Done